src/Server/Resource/DoctrineResource.php

```diff
@@ -543,20 +548,21 @@ public function patch($id, $data)
         }
             // @codeCoverageIgnoreEnd
 
-        // Hydrate entity with patched data
-        $this->getHydrator()->hydrate((array) $data, $entity);
```

Comment:
@nobesnickr and I are discussing adding $data to the events now: https://github.com/zfcampus/zf-apigility-doctrine/issues/183